### ISSUE TITLE
GCS Backend Named States Not Supported

### ISSUE THREAD

**Issue by FellowHobbyist** [hashicorp/terraform#14909]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Summary
Attempting to use GCS Bucket (Google Cloud Storage) as a configured Backend for a Terraform Project, observing the following errors when attempting to Manage, Plan, or Apply using a Name State (i.e. terraform.env)

`named states not supported`

If we manually add the named state via .terraform/environment we observe the following error:

`Error loading state: named states not supported`

As a result, only the Default state is managed and stored on the backend.  Have used S3 as a configured Backend to manage named states for a Terraform Project that spans multiple environments (e.g. dev, tst, prd), and GCS Backend is not functioning as expected based on documentation... https://www.terraform.io/docs/backends/types/gcs.html

### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

$ terraform --version
Terraform v0.9.6

### Affected Resource(s)
Please list the resources as a list, for example:
- gcs backend (Google Cloud Storage)

### Terraform Configuration Files
```
terraform {
  backend "gcs" {
    bucket  = "example-terraform"
    path    = "app_gcs_backend_test"
    project = "example-terraform"
  }
}
```

### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

```
$ terraform env list
named states not supported
```

```
$ terraform env new dev
named states not supported
```

### Expected Behavior
What should have happened?

Should be able to create and manage named states (i.e. environments) when using GCS Backend
```
$ terraform env list
  default
* dev
```

### Actual Behavior
What actually happened?

`ERROR: named states not supported`
`ERROR: Error loading state: named states not supported`

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `create a sample terraform project folder with a gcs backend`
2. `initialize the gcs backend`
3. `Attempt sample commands for managing named states (i.e. environments)`
- `terraform env list`
- `terraform env new <environment>`
4. `Attempt to PLAN or APPLY the provisioning of a resource (e.g. AWS Resource) with GCS Backend using ${terraform.env} Variables... you will notice the error on the console and the state will be DEFAULT vs a NAMED STATE (e.g. dev, tst, prd)

**Comment by tombuildsstuff** [comment]
Hi @FellowHobbyist 

Thanks for opening this issue :)

Currently only the Consul and S3 Backends support Named States - the other Backends (including GCS) only provide Remote State support and need to be updated to support this functionality.

I can't see a PR for this at the moment - as such I'm going to mark this as a feature request for the moment.

Thanks!

**Comment by mattKaczorowski** [comment]
:+1: GCS support would be :rainbow:

**Comment by ljfranklin** [comment]
Looks like the gcs backend was added as part of Terraform 0.11.0, this can probably be closed.

**Comment by tombuildsstuff** [comment]
@ljfranklin I've just checked the changelog and can confirm this shipped in Terraform 0.11.0 - as such I'm going to close this issue.

Thanks!
